boilerplate/openshift/golang-osd-operator/codecov.sh

```diff
@@ -42,4 +42,9 @@ export CI_BUILD_URL="${JOB_LINK}"
 export CI_BUILD_ID="${JOB_NAME}"
 export CI_JOB_ID="${BUILD_ID}"
 
-bash <(curl -s https://codecov.io/bash) -Z -K -f "${COVER_PROFILE}" -r "${REPO_OWNER}/${REPO_NAME}" ${REF_FLAGS}
+if [[ -n "${ARTIFACT_DIR:-}" ]]; then
+       curl -s https://codecov.io/bash -o "${ARTIFACT_DIR}/codecov.sh"
+       bash <(cat "${ARTIFACT_DIR}/codecov.sh") -Z -K -f "${COVER_PROFILE}" -r "${REPO_OWNER}/${REPO_NAME}" ${REF_FLAGS}
+else
```

Comment:
Much as I dig the idea of a failsafe, this opens the possibility of us never storing the script and not finding out about it until we need it. I think it might be better just to blow up, forcing the user to figure out what’s wrong and fix it.

Reply:
This was meant to handle local job types. I've now added additional checks.